src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
```

Comment:
PullProofAI

Consider injecting JwtService into the LoggerMiddleware constructor instead of creating a new instance internally.
```typescript
1: import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
2: import { Request, Response, NextFunction } from 'express';
3: import { JwtService } from '@nestjs/jwt';

5: @Injectable()
6: export class LoggerMiddleware implements NestMiddleware {
7:   private readonly logger = new Logger();
8:   constructor(private readonly jwt: JwtService) {}

10:   use(req: Request, res: Response, next: NextFunction) {
11:     const token = req.headers?.authorization?.split(' ')[1] || '';
12:     const userDetail = this.jwt.decode(token) || {};
```
By injecting `JwtService` in the constructor, you can improve testability and make the code more modular.



<!-- This is an auto-generated comment by PullProofAI -->